### PR TITLE
Create servlet to handle feed page match decisions

### PR DESCRIPTION
**Pull request by mivdittamo**
### What is a quick description of the change?

- Add a servlet to update match information stored in datastore by taking in a user's feed decision regarding a potential match

- Add functionality to find mutual matches and store that information as well

- Add tests to check functionality of the new servlet

### Is this fixing an issue?

none

### Are there more details that are relevant?

none

### Check lists (check `x` in `[ ]` of list items)

- [ x ] Test written/updating
- [ x ] Tests passing
- [ x ] Coding style (indentation, etc)

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/MatchDecisionsServlet.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Entity;
+import com.google.sps.data.friend_map.UserNode;
+import com.google.sps.data.friend_map.UserFriendsMap;
+import com.google.sps.data.PotentialMatchAlgorithm;
+import com.google.sps.data.MatchInformation;
+
+/**
+* Servlet to handle requests to update match decision information.
+*/
```
Comment (jessdejong) [inline src/main/java/com/google/sps/servlets/MatchDecisionsServlet.java:41]: nit: start these with an additional space, to line up the first asterisks.

Comment (jessdejong) [reply src/main/java/com/google/sps/servlets/MatchDecisionsServlet.java:41]: same for the rest too

File: src/test/java/com/google/sps/servlets/MatchDecisionsServletTest.java [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+    when(mockRequest.getParameter(USER_ID_REQUEST_PARAM)).thenReturn(TEST_USER_1_ID);
+    when(mockRequest.getParameter(POTENTIAL_MATCH_REQUEST_PARAM)).thenReturn(TEST_USER_2_ID);
+    when(mockRequest.getParameter(DECISION_REQUEST_PARAM)).thenReturn(FRIENDED_DECISION);
+
+    addTestMatchInfoToDatastore(datastore, TEST_USER_1_ID, ImmutableList.of(TEST_USER_2_ID),
+      ImmutableList.of(), ImmutableList.of(), ImmutableList.of());
+    addTestMatchInfoToDatastore(datastore, TEST_USER_2_ID, ImmutableList.of(TEST_USER_1_ID),
+      ImmutableList.of(), ImmutableList.of(), ImmutableList.of());
+
+    servletUnderTest.doPost(mockRequest, mockResponse);
```
Comment (jessdejong) [inline src/test/java/com/google/sps/servlets/MatchDecisionsServletTest.java:114]: How do you use a mock/local datastore in the test? It looks like you're just using the actual datastore implementation.

Comment (jessdejong) [reply src/test/java/com/google/sps/servlets/MatchDecisionsServletTest.java:114]: Daniel suggested (on my most recent PR) too make my blobstore and datastore instances in my servlet package-private, and then reassign them in the setUp method of my testing file.

File: src/main/java/com/google/sps/servlets/MatchDecisionsServlet.java (hunk 1 above)
Comment (mivdittamo) [reply src/main/java/com/google/sps/servlets/MatchDecisionsServlet.java:41]: Got it!

File: src/test/java/com/google/sps/servlets/MatchDecisionsServletTest.java (hunk 2 above)
Comment (mivdittamo) [reply src/test/java/com/google/sps/servlets/MatchDecisionsServletTest.java:114]: I declared datastore as a private variable, and then I'm initializing it in the setUp() method. 

I'll take a look into the package-private instances!

File: src/test/java/com/google/sps/servlets/MatchDecisionsServletTest.java [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+
+    helper.setUp();
+
+    servletUnderTest = new MatchDecisionsServlet();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    servletUnderTest.datastore = datastore;
+
```
Comment (jessdejong) [inline src/test/java/com/google/sps/servlets/MatchDecisionsServletTest.java:92]: nit: remove newline

**Pull request merged**